### DISCUSSION TITLE
(Bug or feature?) Renovate config on-boarding PR extends from `username/renovate-config` automatically

### DISCUSSION THREAD

**Discussion by HonkingGoose** [renovatebot/renovate#9839] [Request Help] [answered]
I'm not sure if this behavior is a bug or intended behavior, so I'm asking the Renovate code wizards. 🧙

## Situation:

I am helping somebody with a new Renovate config with a throwaway repo, where I'm showing them the initial "default" Renovate bot on-boarding PR and PR message.

I also have a fork of a `renovate-config` repository from somebody else, into my honkinggoose namespace: `honkinggoose/renovate-config` (based on `other-namespace/renovate-config`). I do not want to use any of this configuration on my stuff. 😄 

All of this is happening on the GitHub platform.

## Feature or bug?

The Renovate on-boarding PR (on the throwaway repository) proposes this `renovate.json` file:

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "local>HonkingGoose/renovate-config"
  ]
}
```

Where I would have expected just a boilerplate `renovate.json` similar to this:

```json
{
  "extends": ["config:base"]
}
```

## Question:

Does Renovate "autodiscover" a `renovate-config` repo in the users namespace and use that in onboarding PRs?

## Link to relevant docs:

Link to relevant Renovate Docs: https://docs.renovatebot.com/config-presets/#github-hosted-presets

**Comment by HonkingGoose** [comment]
Relevant snippet from the logs of the Renovate on-boarding run:

```
DEBUG: Need to create onboarding PR
DEBUG: createOnboardingBranch()
DEBUG: Checking if this org/owner has a default Renovate preset which can be used.
DEBUG: onboarding config
{
  "config": {
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "extends": [
      "local>HonkingGoose/renovate-config"
    ]
  }
}
DEBUG: Creating onboarding branch
```

So based on this, I think the automatic extending of a `renovate-config` repo in the users namespace is wanted/expected behavior?

**Reply by rarkins** [reply]
Yes, it is expected. Can you work out the best place to document it? Maybe in the presets section too?

**Reply by HonkingGoose** [reply]
> Yes, it is expected.

This intended behavior just caught me by surprise.

1. Fork other users `renovate-config` into my own namespace
2. Create new repo to help with configuration
3. End up with weird extend in the new repo

It's probably my fault for using the `renovate-config` fork as-is instead of naming it something else... 🙈 
I've renamed the forked `renovate-config` to something else now to prevent this issue from re-occurring. 😄 

Now I just need to nudge the Renovate bot to re-run the onboarding PR, so that it's showing the normal default settings in the PR body instead of the extended repo settings... 😄 

---

> Can you work out the best place to document it? Maybe in the presets section too?

Will do!

**Reply by HonkingGoose** [reply]
PR #9847 will add this behavior to the documentation.

**Comment by HonkingGoose** [comment, marked as answer]
Copy/pasting the reply from **rarkins**, so that I can mark it as the answer.

> Yes, it is expected.

So this is intended and wanted behavior. 😄
